crmsh/utils.py

```diff
@@ -2460,13 +2461,25 @@ def package_is_installed(pkg, remote_addr=None):
     return rc == 0
 
 
-def ping_node(node):
+def node_reachable_checking(node, ping_count=1, socket_port=22, socket_timeout=3):
     """
-    Check if the remote node is reachable
+    Check if node is reachable by using ping and socket to ssh port
     """
-    rc, _, err = ShellUtils().get_stdout_stderr("ping -c 1 {}".format(node))
-    if rc != 0:
-        raise ValueError("host \"{}\" is unreachable: {}".format(node, err))
+    rc, _, _ = ShellUtils().get_stdout_stderr(f"ping -c {ping_count} {node}")
+    if rc == 0:
+        return True
+    # ping failed, try to connect to ssh port by socket
+    error_msg = f"host \"{node}\" is unreachable"
+    try:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
```

Comment:
Does not work with IPv6. Please use `getaddrinfo` first.

Reply:
I found we already have `utils.check_port_open`, should be reused this time